### PR TITLE
improve error message TS2307

### PR DESCRIPTION
**Pull request by donaldpipowitch**
Fixes "Improve error message for TS2307".

This pull request has no associated issue. I can't count how often I (or others in our office) misunderstood `TS2307`. Today was another day so I finally found the time to make this small PR.

`TS2307` says `Cannot find module '{0}'.` and I (and others) often try to check if the module exists in the file system or not. If you use TypeScript as part of a complex build setup (e.g. within a webpack config) you often think some file cannot be found. But many times everything works fine and what is really missing is a type declaration for this module.

So that is what this PR is about. 🙃

Do I need to touch more files to update the translation or if I need to call a script to generate some files which depend on the message? Thanks!

### CODE REVIEW

File: src/compiler/diagnosticMessages.json [hunk 1]
```diff
@@ -976,7 +976,7 @@
         "category": "Error",
         "code": 2306
     },
-    "Cannot find module '{0}'.": {
+    "Cannot find module '{0}' or cannot find corresponding typings for this module.": {
```
Comment (DanielRosenwasser) [inline src/compiler/diagnosticMessages.json]: typings -> type declarations

Comment (donaldpipowitch) [reply src/compiler/diagnosticMessages.json]: Thank you, fixed that 👍

**Comment by RyanCavanaugh** [comment]
@DanielRosenwasser please follow up with this one; thanks!

**Comment by donaldpipowitch** [comment]
I rebased with the current master to solve conflicts. Is this change fine for you?

**Comment by donaldpipowitch** [comment]
Dear future @donaldpipowitch,

because you always forget how you fixed something, I'll write it down.

1. You tried to enhance an error message. In this case the message with error code `TS2307`.
2. All error messages are located inside `src/compiler/diagnosticMessages.json`. Just search for the code `2307` and change the message from `Cannot find module '{0}'.` to `Cannot find module '{0}' or cannot find corresponding type declarations for this module.`.
3. The messages are used in `snake_case` without special characters in several places inside `src/**`. Search for `Diagnostics.Cannot_find_module_0` and change it to `Diagnostics.Cannot_find_module_0_or_cannot_find_corresponding_type_declarations_for_this_module`.
4. Use VS Codes _"Find and Replace"_ with regular expressions like this to fix the tests:
    - Find: `Cannot find module '(.+?)?'\.`
    - Replace: `Cannot find module '$1' or cannot find corresponding type declarations for this module.`
    - In: `tests/**`

**Comment by donaldpipowitch** [comment]
CI has finished. @RyanCavanaugh or @DanielRosenwasser , it's fixed again 😍

**Comment by DanielRosenwasser** [comment]
It's just on me; I don't check GitHub notifications as often as I used to but just update the baselines and we can get it in.

**Comment by donaldpipowitch** [comment]
No problem, thank you very much. I'll fix this and give you a ping!

**Comment by donaldpipowitch** [comment]
@DanielRosenwasser , it's green again. Thank you!

File: src/compiler/diagnosticMessages.json [hunk 2]
```diff
@@ -1040,7 +1040,7 @@
         "category": "Error",
         "code": 2306
     },
-    "Cannot find module '{0}'.": {
+    "Cannot find module '{0}' or cannot find corresponding type declarations for this module.": {
```
Comment (sheetalkamat) [inline src/compiler/diagnosticMessages.json]: @DanielRosenwasser  Shouldn't this be? `Cannot find module '{0}' or its corresponding type declarations.`

Comment (DanielRosenwasser) [reply src/compiler/diagnosticMessages.json]: I like that one better since it's a lot cleaner without giving up details 👍

**Comment by donaldpipowitch** [comment]
Thank you for the feedback!

**Comment by donaldpipowitch** [comment]
Conflicts again 😭

**Comment by donaldpipowitch** [comment]
@DanielRosenwasser or @sheetalkamat : Can we merge this? ❤️

**Comment by AndyRoyal** [comment]
TS2307  I rember you

**Comment by sandersn** [comment]
Looks like this just needs a merge to be ready to go. @donaldpipowitch I'll get to it in the next few days if you don't first.

**Comment by sheetalkamat** [comment]
@sandersn the unit tests folder has changed so the recent merge is bringing back some of the old files ..

**Comment by sandersn** [comment]
@sheetalkamat how did they change? The one that I was aware of was moving src/compiler/testRunner/... up to src/testRunner/...
Are there others? I'll look for incorrectly re-added files there in the meantime.

**Comment by sandersn** [comment]
ok, I think the 3 files in unittests, plus prettier's user test baseline, are all that were incorrectly re-added.

**Pull request merged**